pkg/gohai/filesystem/filesystem_nix.go

```diff
@@ -0,0 +1,135 @@
+// This file is licensed under the MIT License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2014-present Datadog, Inc.
+
+//go:build darwin || linux
+// +build darwin linux
+
+package filesystem
+
+import (
+	"strings"
+	"syscall"
+
+	"github.com/moby/sys/mountinfo"
+)
+
+// These FS types should be excluded from listing
+var IgnoredFSTypes = map[string]struct{}{
+	"autofs":      {},
+	"debugfs":     {},
+	"devfs":       {},
+	"devpts":      {},
+	"fuse.portal": {},
+	"fusectl":     {},
+	"ignore":      {},
+	"kernfs":      {},
+	"none":        {},
+	"proc":        {},
+	"subfs":       {},
+	"mqueue":      {},
+	"rpc_pipefs":  {},
+	"sysfs":       {},
+}
+
+// These FS types are known to be remote
+var RemoteFSTypes = map[string]struct{}{
+	"acfs":       {},
+	"afs":        {},
+	"auristorfs": {},
+	"coda":       {},
+	"fhgfs":      {},
+	"gpfs":       {},
+	"ibrix":      {},
+	"ocfs2":      {},
+	"vxfs":       {},
+}
+
+func getFileSystemInfo() ([]MountInfo, error) {
+	return getFileSystemInfoWithMounts(nil, true)
+}
+
+// Internal method to help testing with test mounts
+// If ignoreEmpty is true, ignore zero-sized filesystems, like `df` does (unless using -a)
+func getFileSystemInfoWithMounts(initialMounts []*mountinfo.Info, ignoreEmpty bool) ([]MountInfo, error) {
+	var err error
+	mounts := initialMounts
+
+	if mounts == nil {
+		mounts, err = mountinfo.GetMounts(nil)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var mountInfos = make([]MountInfo, 0, len(mounts))
+
+	for _, mount := range mounts {
+		if isExcludedFS(mount.FSType, mount.Source, true) {
+			continue
+		}
+
+		var stat syscall.Statfs_t
+
+		sizeKB := uint64(0)
+		if err := syscall.Statfs(mount.Mountpoint, &stat); err == nil {
+			sizeKB = (stat.Blocks * uint64(stat.Bsize)) / 1024
+		}
+
+		if ignoreEmpty && sizeKB == 0 {
+			continue
+		}
+
+		// Skip mounts that seem to be missing data
+		if mount.Source == "" || mount.Source == "none" || mount.FSType == "" || mount.Mountpoint == "" {
+			continue
+		}
+
+		mountInfo := MountInfo{
+			Name:      mount.Source,
+			SizeKB:    sizeKB,
+			MountedOn: mount.Mountpoint,
+		}
+		mountInfos = append(mountInfos, mountInfo)
+	}
+
+	return mountInfos, nil
+}
+
+func isExcludedFS(fsType string, fsSource string, localOnly bool) bool {
```

Comment:
See https://github.com/coreutils/gnulib/blob/master/lib/mountlist.c#L171-L201 for the implementation in `gnulib`

Reply:
Note: the well-named [treat-devtmpfs-and-squashfs-as-dummy-filesystems.patch](https://github.com/DataDog/datadog-agent/files/12112306/treat-devtmpfs-and-squashfs-as-dummy-filesystems.patch) adds `devtmpfs` and `squashfs` to the excluded list on Ubuntu (and possibly other Linux OS).

Reply:
Done in https://github.com/DataDog/datadog-agent/pull/18117/commits/82aec9f215afce04cebd48d33e0c71c1562397a3